### PR TITLE
Make the missing error in rootcling a warning

### PR DESCRIPTION
**Pull request by Teemperor**
This code assumed that rootcling doesn't return zero when we
print an error, but this is not the case. Let's downgrade it to
a warning that we actually still return a zero exit code.

Also the method now just returns true instead of writing 'true'
twice to the same result variable and then returning that.

### CODE REVIEW

File: core/dictgen/src/rootcling_impl.cxx [hunk 1]
```diff
@@ -994,20 +994,16 @@ bool CheckClassDef(const clang::RecordDecl &cl, const cling::Interpreter &interp
...
       ROOT::TMetaUtils::GetQualifiedName(qualName, cl);
       const char *qualName_c = qualName.c_str();
-      ROOT::TMetaUtils::Error(qualName_c,
-                              "%s inherits from TObject but does not have its own ClassDef\n",
-                              qualName_c);
-      // We do want to always output the message (hence the Error level)
-      // but still want rootcling to succeed.
-      result = true;
+      ROOT::TMetaUtils::Warning(qualName_c,
+                                "%s inherits from TObject but does not have its own ClassDef\n",
```
Comment (Axel-Naumann) [inline core/dictgen/src/rootcling_impl.cxx]: Now that it's a warning we should state why people have to care. Can you rephrase to `"The data members of %s will not be stored, because it inherits from TObject but does not have its own ClassDef."`?

Comment (Teemperor) [reply core/dictgen/src/rootcling_impl.cxx]: Done!

**Comment by Axel-Naumann** [comment]
@dpiparo @pcanal please review post merge!
